bus/prometheus.go

```diff
@@ -0,0 +1,857 @@
+package bus
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/alerts"
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/hostdb"
+	"go.sia.tech/renterd/internal/prometheus"
+	"go.sia.tech/renterd/wallet"
+)
+
+type AlertsResp []alerts.Alert
+
+// PrometheusMetric returns Prometheus samples for the hosts alerts.
+func (a AlertsResp) PrometheusMetric() (metrics []prometheus.Metric) {
+	for _, alert := range a {
+		metrics = append(metrics, prometheus.Metric{
+			Name: "renterd_alert",
+			Labels: map[string]any{
+				"id":        alert.ID,
+				"severity":  alert.Severity.String(),
+				"message":   alert.Message,
+				"timestamp": alert.Timestamp,
+			},
+			Value: 1,
+		})
+	}
+	return
+}
+
+type BucketResp []api.Bucket
+
+func (b BucketResp) PrometheusMetric() (metrics []prometheus.Metric) {
+	for _, bucket := range b {
+		metrics = append(metrics, prometheus.Metric{
+			Name: "renterd_bucket",
+			Labels: map[string]any{
+				"name": bucket.Name,
+				"publicReadAccess": func() float64 {
+					if bucket.Policy.PublicReadAccess {
+						return 1
+					}
+					return 0
+				}(),
+				"createdAt": bucket.CreatedAt.String(),
+			},
+			Value: 1,
+		})
+	}
+	return
+}
+
+type ConsensusStateResp api.ConsensusState
+
+func (c ConsensusStateResp) PrometheusMetric() (metrics []prometheus.Metric) {
+	metrics = append(metrics, prometheus.Metric{
+		Name: "renterd_consensus_state_synced",
+		Value: func() float64 {
+			if c.Synced {
+				return 1
+			}
+			return 0
+		}(),
+	})
+	metrics = append(metrics, prometheus.Metric{
+		Name:  "renterd_consensus_state_chain_index_height",
+		Value: float64(c.BlockHeight),
+	})
+	return
+}
+
+type ContractsResp []api.ContractMetadata
```

Comment:
What happens if you define `PrometheusMetric` on `api.ContractMetadata`. Wouldn't it automatically work for slices of the type too? This is a little add and when we move this `PrometheusMetric` methods to the `api` package it would be nice if we could avoid adding custom types for everything we return slices of.

Reply:
im not sure